### PR TITLE
Throws ValueError when leap second is detected

### PR DESCRIPTION
**Pull request by ritwickdsouza**
References #4983

### CODE REVIEW

**Comment by MSeifert04** [comment]
Thank you for the pull request. 👍 

It's not enforced but you should check the indentation to be consistent:

```
                raise ValueError('Time {} is within a leap second but dateime '
                                 'does not support leap seconds'
                                 .format((iy, im, id, ihr, imin, isec, ifracsec)))
```

This also needs a changelog entry (probably for 1.3.1 but wait until someone sets a milestone for the PR) and a test.

**Comment by ritwickdsouza** [comment]
@MSeifert04 Did that. I'll keep it in mind. :)

File: astropy/time/formats.py [hunk 1]
```diff
@@ -603,6 +603,10 @@ def to_value(self, timezone=None, parent=None):
                              op_dtypes=7*[iys.dtype] + [np.object])
 
         for iy, im, id, ihr, imin, isec, ifracsec, out in iterator:
+            if isec >= 60:
+                raise ValueError('Time {} is within a leap second but dateime '
```
Comment (saimn) [inline astropy/time/formats.py]: typo: date**t**ime

**Comment by ritwickdsouza** [comment]
@saimn Thanks. Fixed that.

**Comment by ritwickdsouza** [comment]
Any changes that need to be made for this PR ?

**Comment by pllim** [comment]
Bug fix needs a change log entry under 1.3.1 section. See other entries for the required format. You can add `[skip ci]` to your *commit message* to skip CI from running on that commit.

The maintainers for this module is @taldcroft and @mhvk .

**Comment by mhvk** [comment]
Sorry to come late to this: I'm not sure this actually needs a change-log entry, since really what it does is give a more informative error message (which is great!). What should be done, however, is to add a test case, so that we can be sure the code works as intended. For this purpose, maybe just use the example from #4983. I'd probably put the test in `astropy/time/tests/test_basic`, in the `test_to_datetime` function. @ritwickdsouza - let me know if you have time for this; otherwise I can do it.

**Comment by ritwickdsouza** [comment]
@mhvk I will do the tests

**Comment by ritwickdsouza** [comment]
@mhvk Is the test alright or should it be more specific ?

File: astropy/time/tests/test_basic.py [hunk 2]
```diff
@@ -1056,6 +1056,9 @@ def test_to_datetime():
         assert tz.tzname(dt) == tz_dt.tzname()
     assert np.all(time == forced_to_astropy_time)
 
+    with pytest.raises(ValueError):
+        Time('2015-06-30 23:59:60.000').to_datetime()
```
Comment (bsipocz) [inline astropy/time/tests/test_basic.py]: Could you test for getting the new error message? This test would pass even without your PR as even then a ValueError was raised.

Comment (ritwickdsouza) [inline astropy/time/tests/test_basic.py]: Noted. How do i get `.format((iy, im, id, ihr, imin, isec, ifracsec))` part to compare ?

Comment (mhvk) [inline astropy/time/tests/test_basic.py]: You could copy & paste the message you get from an interactive test run, or just check part of the message, as in
```
assert 'datetime does not support' in str(message)
```
I personally prefer the latter -- I don't think it is all that important to ensure a test message is always identical, we just want to ensure we get the one that you put it!

File: astropy/time/tests/test_basic.py [hunk 3]
```diff
@@ -1076,6 +1079,9 @@ def test_to_datetime_pytz():
         assert tz.tzname(dt) == tz_dt.tzname()
     assert np.all(time == forced_to_astropy_time)
 
+    with pytest.raises(ValueError):
```
Comment (mhvk) [inline astropy/time/tests/test_basic.py]: I don't think we need this second test, or does it really do something different?

Comment (ritwickdsouza) [inline astropy/time/tests/test_basic.py]: Removed the test

**Comment by mhvk** [comment]
@ritwickdsouza - it all looks good, but there were two test failures still: one is spurious, and the other due to an extra whitespace. I fixed the latter, and will merge if all tests now pass. Thanks!

**Comment by ritwickdsouza** [comment]
@mhvk Great! My first Pull Request getting merged. :)

**Comment by mhvk** [comment]
OK, all tests now passed, so merging. Thanks, @ritwickdsouza!

**Pull request merged**